docker/context/context.py

```diff
@@ -16,30 +16,42 @@ def __init__(self, name, orchestrator=None, host=None, endpoints=None,
         if not name:
             raise Exception("Name not provided")
         self.name = name
+        self.context_type = None
         self.orchestrator = orchestrator
+        self.endpoints = {}
+        self.tls_cfg = {}
+        self.meta_path = "IN MEMORY"
+        self.tls_path = "IN MEMORY"
+
         if not endpoints:
+            # set default docker endpoint if no endpoint is set
             default_endpoint = "docker" if (
                 not orchestrator or orchestrator == "swarm"
                 ) else orchestrator
+
             self.endpoints = {
                 default_endpoint: {
                     "Host": get_context_host(host, tls),
                     "SkipTLSVerify": not tls
                 }
             }
-        else:
-            for k, v in endpoints.items():
-                ekeys = v.keys()
-                for param in ["Host", "SkipTLSVerify"]:
-                    if param not in ekeys:
-                        raise ContextException(
-                            "Missing parameter {} from endpoint {}".format(
-                                param, k))
-            self.endpoints = endpoints
+            return
 
-        self.tls_cfg = {}
-        self.meta_path = "IN MEMORY"
-        self.tls_path = "IN MEMORY"
+        # check docker endpoints
+        for k, v in endpoints.items():
+            if not isinstance(v, dict):
+                # unknown format
+                raise ContextException("""Unknown endpoint format for
+                    context {}: {}""".format(name, v))
+
+            self.endpoints[k] = v
+            if k != "docker":
+                continue
```

Comment:
I wonder if `docker-py` should ignore these, or just load them. I'm not too familiar with the code, buta context has _some_ (semi-)strict fields, and other metadata that could be "anything". I think it should be ok to _load_ these, as long as they're not used to try to _connect_ with them.

(basically; if this library is just to _parse_ / _load_ contexts, it should not be too opinionated)

 WDYT?

Reply:
They are loaded,  for docker endpoints we only verify that we have a Host and SkipTLSVerify fields set (otherwise we set some defaults).  For cloud endpoints we don't check anything.

Reply:
🤦 I see it now, yes. I read the `continue` as "skip this endpoint", but it's loaded the line above, and only the special handling is skipped